mono/mini/interp/interp.c

```diff
@@ -282,10 +271,11 @@ pop_frame (ThreadContext *context, InterpFrame *frame)
 {
 	if (frame->stack)
 		frame_stack_pop (&context->data_stack, frame->data_frag, frame->stack);
-	frame_stack_pop (&context->iframe_stack, frame->iframe_frag, frame);
-}
 
-#define interp_exec_method(frame, context, error) interp_exec_method_full ((frame), (context), NULL, error)
+	InterpFrame* const parent = frame->parent;
+	if (parent)
```

Comment:
why do we need this ? Aren't the frames linked from allocation

Reply:
Originally did link them at allocation, yes.
With the comment "Not free currently, but will be when allocation occurs."
I had problems -- not sure if related.
Main problem seems to have been via `interp_run_finally`, which took me a while to realize.
So I moved it here, and had other stuff, in the meantime.
I can try there instead.
That would be a small optimization, since pop_frame is more frequent than alloc_frame-replacement.

Maybe a separate PR to leave this working..

Reply:
Let's see if https://github.com/mono/mono/pull/19007 works. I think that is what you mean.

Reply:
I thought I saw them linked also on allocation, maybe on old version of the pr. But yeah, I'd prefer it that way as a small optimization

Reply:
Agreed & yes it was there, trying it again that way.

Reply:
https://github.com/mono/mono/pull/19007 appears to work so closing this.